### ISSUE TITLE
Replace JSON wrapper with multi-level publish topics

### ISSUE THREAD

**Issue by lenny-intel** [edgexfoundry/device-mqtt-go#282]
Putting this out here for discussion. If done post 2.0 release we code implement with a feature flag.

Currently the incoming data (async) on the simple `DataTopic`  topic has the data wrapped in JSON as follows:
```
{
  "name" : "<device name>",
  "cmd" : "<command name>",
  "<command name>" : "<data>"
}
```
This could be accomplished with mult-level topic similar to how the messages are now being published to the MessageBus.
```
incoming/data/<devicename>/<commandname>
```
Incoming subscribe topic becomes `/incoming/data/#`
Then the data published is just the reading, raw not wrapped.
Code parses the received topic to extract the `devicename` and `commandname`

Similar approach for publishing command the command topic would be
```
command/<devicename>/<commandname>/<method>/<UUDI> 
```
Device subscribes to `command/<devicename>/#`
and for receiving command response
```
command/response/<uuid>
```
Code parses the received topic to extract the `uuid` and the data received is the reading(s), not wrapped, for Get and just an ack for Put
Response topic becomes `command/response/#`

**Comment by lenny-intel** [comment]
@cloudxxx8 , @hahattan , @iain-anderson, @tonyespy  , thoughts?

**Comment by cloudxxx8** [comment]
@lenny-intel the original design was made several years ago (in Java version), and we moved what it was to the golang version.  I feel it was just a sample implementation.  To make this more useful, we have to find some real MQTT Devices to retrieve real requirements before spending more time.
We don't know whether there will be a <uuid> field, or whether the topics of Devices can be configured as what we designed.  Maybe, those topics should be defined in the Device Profile.

**Comment by SteveOss** [comment]
My preference WRT to Topic usage is to keep the number of topics per device to a tractable number i.e. use JSON wrapping for request/reply. However I have seen Topics used at a much finer level of granularity where a Topic per resource was used. We have been working with some real BACnet devices recently, where a typical device might have 5000 current value points (e.g. analog input), but each of which could have at least a dozen properties (units etc) which themselves are represented as properties. So here we are talking about 60000 properties per device, so I am not sure a resource/topic mapping here is the best solution. Might be also worth taking a look at how the Eclipse SparkPlug specification defines the usage of MQTT topics https://sparkplug.eclipse.org

**Comment by antoniomtz** [comment]
To support @lenny-intel 's proposal, we have implemented a couple of use cases in the past where we had to do extra work to transform the MQTT payload coming from the device to comply with that specific EdgeX  JSON schema.
For instance, creating a simple DL Streamer pipeline to publish inference data to EdgeX, we had to add an extra script in python to transform the output to that specific EdgeX schema.
If we just specify the right topic as `/incoming/data/<devicename>/<commandname>`  and publish the raw data into it, it will simplify things and eliminate extra work.
Similarly, when working with a RFID vendor, we had to develop an extra service to receive the data, transform the schema and republish it to the MQTT broker.
We would've loved to simply publish the data from the RFID vendor to EdgeX directly.
